hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NSSummaryEndpoint.java

```diff
@@ -0,0 +1,1061 @@
+      return Response.ok(quotaUsageResponse).build();
+    }
+
+    String normalizedPath = normalizePath(path);
+    String[] names = parseRequestPath(normalizedPath);
+    EntityType type = getEntityType(normalizedPath, names);
+
+    if (type == EntityType.ROOT) {
+      List<OmVolumeArgs> volumes = listVolumes();
+      List<OmBucketInfo> buckets = listBucketsUnderVolume(null);
+      long quotaInBytes = 0L;
+      long quotaUsedInBytes = 0L;
+
+      for (OmVolumeArgs volume: volumes) {
+        final long quota = volume.getQuotaInBytes();
+        assert(quota >= -1L);
+        if (quota == -1L) {
+          // If one volume has unlimited quota, the "root" quota is unlimited.
+          quotaInBytes = -1L;
+          break;
+        }
+        quotaInBytes += quota;
+      }
+      for (OmBucketInfo bucket: buckets) {
+        long bucketObjectId = bucket.getObjectID();
+        quotaUsedInBytes += getTotalSize(bucketObjectId);
+      }
+
+      quotaUsageResponse.setQuota(quotaInBytes);
+      quotaUsageResponse.setQuotaUsed(quotaUsedInBytes);
+    } else if (type == EntityType.VOLUME) {
+      List<OmBucketInfo> buckets = listBucketsUnderVolume(names[0]);
+      String volKey = omMetadataManager.getVolumeKey(names[0]);
+      OmVolumeArgs volumeArgs =
+          omMetadataManager.getVolumeTable().getSkipCache(volKey);
+      long quotaInBytes = volumeArgs.getQuotaInBytes();
+      long quotaUsedInBytes = 0L;
+
+      // Get the total data size used by all buckets
+      for (OmBucketInfo bucketInfo: buckets) {
+        long bucketObjectId = bucketInfo.getObjectID();
+        quotaUsedInBytes += getTotalSize(bucketObjectId);
+      }
+      quotaUsageResponse.setQuota(quotaInBytes);
+      quotaUsageResponse.setQuotaUsed(quotaUsedInBytes);
+    } else if (type == EntityType.BUCKET) {
+      String bucketKey = omMetadataManager.getBucketKey(names[0], names[1]);
+      OmBucketInfo bucketInfo = omMetadataManager
+          .getBucketTable().getSkipCache(bucketKey);
+      long bucketObjectId = bucketInfo.getObjectID();
+      long quotaInBytes = bucketInfo.getQuotaInBytes();
+      long quotaUsedInBytes = getTotalSize(bucketObjectId);
+      quotaUsageResponse.setQuota(quotaInBytes);
+      quotaUsageResponse.setQuotaUsed(quotaUsedInBytes);
+    } else if (type == EntityType.UNKNOWN) {
+      quotaUsageResponse.setResponseCode(ResponseStatus.PATH_NOT_FOUND);
+    } else { // directory and key are not applicable for this request
+      quotaUsageResponse.setResponseCode(
+          ResponseStatus.TYPE_NOT_APPLICABLE);
+    }
+    return Response.ok(quotaUsageResponse).build();
+  }
+
+  /**
+   * Endpoint that returns aggregate file size distribution under a path.
+   * @param path request path
+   * @return File size distribution response
+   * @throws IOException
+   */
+  @GET
+  @Path("/dist")
+  public Response getFileSizeDistribution(@QueryParam("path") String path)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 50 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.